### DISCUSSION TITLE
[SSL: CERTIFICATE_VERIFY_FAILED] got CERTIFICATE_VERIFY_FAILED when I deployed the contract to rinkeby

### DISCUSSION THREAD

**Discussion by weilainicolas** [smartcontractkit/full-blockchain-solidity-course-py#517] [Q&A] [answered]
w3 = Web3(Web3.HTTPProvider("https://rinkeby.infura.io/v3/c14dc273c46f4fbdb6ddb45ebe9793f5"))
chain_id = 4  
my_address = "0x3562Dba7c0B3e4307Eb3371852498A22A4F84205" 
private_key = os.getenv("PRIVATE_KEY")

requests.exceptions.SSLError: HTTPSConnectionPool(host='rinkeby.infura.io', port=443): Max retries exceeded with url: /v3/c14dc273c46f4fbdb6ddb45ebe9793f5 (Caused by SSLError(SSLCertVerificationError(1, '[SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: self signed certificate in certificate chain (_ssl.c:1123)')))

Anyone has the similar issue?

**Comment by cromewar** [comment, marked as answer]
Hello @weilainicolas this might be happening due to different reasons:

1. First of all make sure to export the .env with the RINKEBY_RPC_URL.
2. If you're using macOS go to Macintosh HD > Applications > Python3.6 folder (or whatever version of python you're using) > double click on "Install Certificates.command" file.
3. does [this](https://community.infura.io/t/request-timeout-with-infura-ropsten-api/642/2) works for you?

Best regards, Cromewar.

**Reply by weilainicolas** [reply]
I installed Python3.8 via anaconda, so I cannot find Python.3.8 under Application.

**Reply by cromewar** [reply]
You would have the same file but on the path you installed the python env with anaconda, anyways I do not recomment to use anaconda for smart contract development as some libraries are not likely to be updated. I highly recommend to use simple python instead with a virtual env.

**Reply by weilainicolas** [reply]
Thanks. I could not find what was going on. So I reboot a window machine with Ubuntu and use it solely for development. It is a more sustainable way.
